api/lock.go

```diff
@@ -123,7 +123,8 @@ func (c *Client) LockOpts(opts *LockOptions) (*Lock, error) {
 
 // Lock attempts to acquire the lock and blocks while doing so.
 // Providing a non-nil stopCh can be used to abort the lock attempt.
-// Returns a channel that is closed if our lock is lost or an error.
+// If the lock is aborted, it returns a channel that is nil. If the lock is lost
+// or an error occurred, it returns a channel that is closed.
```

Comment:
This is not quite accurate - the chan is returned open but then later closed when the lock is lost or the connection lost.

I think the clearest way to word this would be something like:

```
// If the lock is successfully acquired (err is nil) then a chan is returned that will be 
// closed when the lock is lost or becomes indeterminate (due to connection failure 
// for example). If an err is returned or stopCh is closed, the returned chan is nil.